docker/Dockerfile

```diff
@@ -0,0 +1,83 @@
+FROM ubuntu:22.04
+MAINTAINER Weina Ji <weina.ji@epfl.ch>
+
+# The default shell for the RUN instruction is ["/bin/sh", "-c"].
+# Using SHELL instruction to change default shell for subsequent RUN instructions
+SHELL ["/bin/bash", "-c"]
+
+# Create a directory for installation
+ARG WORKDIR=/opt/software
+RUN mkdir $WORKDIR
+
+# Install needed libs
+RUN apt-get --yes -qq update \
+ && apt-get --yes -qq upgrade \
+ && apt-get --yes -qq install \
+                      g++ \
+                      gcc \
+                      python3.10 \
+                      python3-pip \
+                      python3-venv \
+                      git \
+                      cmake \
+                      libreadline-dev \
+                      wget \
+                      mpich libmpich-dev libhdf5-mpich-dev hdf5-tools
```

Comment:
Some of the below are nitpicks (or preferences)! See which ones you think make sense:

* Should we install specific versions/tags of software here? e.g. I am wondering if installing neuron master is a good idea or if it should be some tag from spack recipe. Same for others.
* Just nitpicks:
   * installing all packages in some common install prefix would be nicer? e.g. `<some-dir>/install`. And then at the end we have `PATH` and `PYTHONPATH` added for this one directory and its done.
   * Maybe grouping some of the instructions would be nicer? e.g. all apt installed packages at the top. Set PATH/PYTHONPATH at the end once (to common install dir).

Reply:
> Should we install specific versions/tags of software here? e.g. I am wondering if installing neuron master is a good idea or if it should be some tag from spack recipe. Same for others.

How about using the latest tag of each repo? It is easier than getting the version from the spack recipe. The tags of libsonata, libsonatareport are always updated in the spack. Only the neuron tag is behind its spack verson, but I think it is safer to use the tag version tag.

Reply:
sounds good!